### PR TITLE
PostgreSQL high availability

### PR DESCRIPTION
**Pull request by lawrencejones**
Hey all,

This change continues on from #695 to address the review comments there and finish off the high availability support.

The only changes from #695 are the addition of an integration test and some fixing up of the sqladmin package. Also added some comments to clarify that the 1st Gen instances will required diff suppressing, and a default is provided server side when availability zone isn't given.

@katzj, I hope you don't mind but we really needed this at GoCardless a few months back!

---

Integration test output is:

```
λ ~> make testacc TEST=./google TESTARGS='-run=TestAccGoogleSqlDatabaseInstance_highAvailability' GOOGLE_USE_DEFAULT_CREDENTIALS=true
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccGoogleSqlDatabaseInstance_highAvailability -timeout 120m
=== RUN   TestAccGoogleSqlDatabaseInstance_highAvailability
--- PASS: TestAccGoogleSqlDatabaseInstance_highAvailability (561.98s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 562.009s
```

### CODE REVIEW

**Comment by lawrencejones** [comment]
Hey @danawillow, this PR should cover the comments you made on #695. Can you give this a second pass please?

File: google/resource_sql_database_instance_test.go [hunk 1]
```diff
@@ -568,7 +606,7 @@ func testAccCheckGoogleSqlDatabaseInstanceEquals(n string,
 			return fmt.Errorf("Error settings.crash_safe_replication mismatch, (%s, %s)", server, local)
 		}
 
-		server = strconv.FormatBool(instance.Settings.StorageAutoResize)
+		server = strconv.FormatBool(*instance.Settings.StorageAutoResize)
```
Comment (danawillow) [inline google/resource_sql_database_instance_test.go]: I'm seeing panics at this line when running tests, can you check it out?

Comment (lawrencejones) [reply google/resource_sql_database_instance_test.go]: Having a look at this now- do you have the test that caused the fatal? Would speed up things on my end, as the few I've been running do not!

Comment (lawrencejones) [reply google/resource_sql_database_instance_test.go]: Think I've got a fix for this, will update the PR once I've confirmed tests pass.

**Comment by lawrencejones** [comment]
Looking a lot healthier this time around:
```
λ ~> make testacc TEST=./google TESTARGS='-run=TestAccGoogleSqlDatabaseInstance_.*' GOOGLE_USE_DEFAULT_CREDENTIALS=true==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccGoogleSqlDatabaseInstance_.* -timeout 120m
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic
=== RUN   TestAccGoogleSqlDatabaseInstance_importBasic3
=== RUN   TestAccGoogleSqlDatabaseInstance_basic
=== RUN   TestAccGoogleSqlDatabaseInstance_basic2
=== RUN   TestAccGoogleSqlDatabaseInstance_basic3
=== RUN   TestAccGoogleSqlDatabaseInstance_dontDeleteDefaultUserOnReplica
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_basic
=== RUN   TestAccGoogleSqlDatabaseInstance_slave
=== RUN   TestAccGoogleSqlDatabaseInstance_highAvailability
=== RUN   TestAccGoogleSqlDatabaseInstance_diskspecs
=== RUN   TestAccGoogleSqlDatabaseInstance_maintenance
=== RUN   TestAccGoogleSqlDatabaseInstance_settings_upgrade
=== RUN   TestAccGoogleSqlDatabaseInstance_settingsDowngrade
=== RUN   TestAccGoogleSqlDatabaseInstance_authNets
=== RUN   TestAccGoogleSqlDatabaseInstance_multipleOperations
--- PASS: TestAccGoogleSqlDatabaseInstance_settingsDowngrade (58.60s)
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic (60.88s)
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_upgrade (109.66s)
--- PASS: TestAccGoogleSqlDatabaseInstance_authNets (73.10s)
--- PASS: TestAccGoogleSqlDatabaseInstance_multipleOperations (130.19s)
--- PASS: TestAccGoogleSqlDatabaseInstance_highAvailability (519.86s)
--- PASS: TestAccGoogleSqlDatabaseInstance_maintenance (422.80s)
--- PASS: TestAccGoogleSqlDatabaseInstance_settings_basic (50.60s)
--- PASS: TestAccGoogleSqlDatabaseInstance_diskspecs (453.66s)
--- PASS: TestAccGoogleSqlDatabaseInstance_basic3 (456.09s)
--- PASS: TestAccGoogleSqlDatabaseInstance_basic (81.38s)
--- PASS: TestAccGoogleSqlDatabaseInstance_basic2 (61.35s)
--- PASS: TestAccGoogleSqlDatabaseInstance_importBasic3 (535.08s)
--- PASS: TestAccGoogleSqlDatabaseInstance_dontDeleteDefaultUserOnReplica (714.24s)
--- PASS: TestAccGoogleSqlDatabaseInstance_slave (898.62s)
PASS
```

**Comment by danawillow** [comment]
Looks good, thanks!

**Pull request merged**